translib/common_app.go

```diff
@@ -891,6 +893,228 @@ func (app *CommonApp) cmnAppCRUCommonDbOpn(d *db.DB, opcode int, dbMap map[strin
 	return err
 }
 
+func deleteFields(existingEntry db.Value, d *db.DB, cmnAppTs *db.TableSpec, tblKey string, tblRw db.Value, deleteEmptyEntry bool) error {
+	var err error
+	log.V(4).Info("DELETE case - fields/cols to delete hence delete only those fields.")
+	/* handle leaf-list merge if any leaf-list exists */
+	tblNm := cmnAppTs.Name
+	resTblRw := checkAndProcessLeafList(existingEntry, tblRw, DELETE, d, tblNm, tblKey)
+	log.V(4).Info("DELETE case - checkAndProcessLeafList() returned table row ", resTblRw)
+	if len(resTblRw.Field) > 0 {
+		if !deleteEmptyEntry {
+			/* add the NULL field if the last field gets deleted && deleteEmpyEntry is false */
+			deleteCount := 0
+			for field := range existingEntry.Field {
+				if resTblRw.Has(field) {
+					deleteCount++
+				}
+			}
+			if deleteCount == len(existingEntry.Field) {
+				nullTblRw := db.Value{Field: map[string]string{"NULL": "NULL"}}
+				log.Info("Last field gets deleted, add NULL field to keep an db entry")
```

Comment:
can we add more context to the log like which field.., and make it log level 4

Reply:
Addressed.